web/client/components/TOC/TOCItemsSettings.jsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const {Row, Col, Nav, NavItem, Glyphicon} = require('react-bootstrap');
+const DockablePanel = require('../misc/panels/DockablePanel');
+const Toolbar = require('../misc/toolbar/Toolbar');
+const tooltip = require('../misc/enhancers/tooltip');
+const NavItemT = tooltip(NavItem);
+const ResizableModal = require('../misc/ResizableModal');
+const Portal = require('../misc/Portal');
+const {head, isObject} = require('lodash');
+const Message = require('../I18N/Message');
+
+module.exports = props => {
+    const {
```

Comment:
This is not generic, too many properties are dependent on specific item info (e.g. chartStyle, getDimension, generalInfoFormat). If it has to be a generic item setting container component, it should only have reference to generic things (e.g. the item node)